browser/resources/brave_new_tab_page_refresh/components/background/safe_area_reporter.ts

```diff
@@ -0,0 +1,50 @@
+/* Copyright (c) 2025 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+import * as React from 'react'
+
+import { RichMediaFrameHandle } from './rich_media_capabilities'
+import { useWidgetLayoutReady } from '../app_layout_ready'
+import { debounce } from '$web-common/debounce'
+
+// Posts a message to the rich media background iframe containing a rectangle
+// that is empty of content and can be used to display interactive elements.
+export function useSafeAreaReporter(frameHandle?: RichMediaFrameHandle) {
```

Comment:
Moved from `background.tsx`. The only change is the usage of the type-safe `RichMediaFrameHandle` interface instead of `IframeBackgroundHandle`.